core/vm/contracts.go

```diff
@@ -1185,14 +1185,19 @@ func (e *esGetBlob) Run(input []byte) ([]byte, error) {
 	}
 
 	kvIndex := new(big.Int).SetBytes(getData(input, 0, 32)).Uint64()
-	off := new(big.Int).SetBytes(getData(input, 32, 32)).Uint64()
-	len := new(big.Int).SetBytes(getData(input, 64, 32)).Uint64()
-	hash := input[96:128]
+	needDecode := false
+	needDecodeInt := new(big.Int).SetBytes(getData(input, 32, 32)).Uint64()
```

Comment:
Why not change needDecode to decodeType, where 0 is raw data, 1 is decoded using 31 bytes (I believe)?